### DISCUSSION TITLE
Contributing: How to retry failed GitHub CI jobs?

### DISCUSSION THREAD

**Discussion by jswxstw** [argoproj/argo-workflows#12840] [Q&A] [answered]
I often encounter issues where certain CI jobs run failed after submitting a PR.
Most of these jobs fail sporadically. So, is it possible to support retrying these jobs?

**Comment by agilgur5** [comment, marked as answer]
Unfortunately, no, not currently. Only those with `write` permissions can retry GitHub CI, i.e. Approver+. This was a larger topic of discussion when there were substantially more test flakes back in [September](https://docs.google.com/document/d/1tznN5LB-KrNkC6dmeIzpKfuyvZWgCJpRuS-E84zseC8/edit#heading=h.k1l4irxjo6vc) (and it was probably discussed during other test flake increases too).

The workaround is to push an empty commit to your branch, which will re-trigger CI. Unfortunately it re-triggers _all_ jobs, not just the failed ones.
The better solution is to add a bot/GH Action that can retry on behalf of Members if they comment `/retest` (similar to the k8s bot), which would more minimally retry failed jobs. I was looking at something similar for automated cherry-picks with `/cherry-pick`: https://github.com/argoproj/argo-workflows/issues/12592#issuecomment-1962996204. Would you be interested in working on adding such an Action? **EDIT**: See #12864

Also the root cause fix is ofc to solve all the flakes, which are often due to unhandled race conditions in the actual code, and sometimes unhandled race conditions in the tests. You've seen #12832 as one example of tracking such test flakes, but there have also been #10807, #9027, #7133 and (many) others in the past.

**Reply by agilgur5** [reply]
>  on behalf of Members

By the way, would you be interested in [becoming a Member of Argoproj](https://github.com/argoproj/argo-workflows/blob/a6b1217ac47fa30e27db4f0a0669b477586fd60d/docs/CONTRIBUTING.md#roles)? You've had some great root cause analyses and found and fixed various bugs, so would be a great addition 🙂 I can be a sponsor and we can probably get other sponsors too

**Reply by Joibel** [reply]
I'd support this too.

**Reply by jswxstw** [reply]
I am responsible for developing a workflow platform based on Argo-Workflow in my company. Therefore, recently I have made some improvements or bug fixes for certain functionalities that we rely on.
In the future, I will be responsible for other tasks, so I won't have as much time to contribute as a member.
However, I am still willing to continue contributing to this project whenever I have free time.😊

**Reply by Joibel** [reply]
There are no expectations of you to contribute in the longer term, you're still welcome to become a member and see where that takes you.

**Reply by agilgur5** [reply]
> I am responsible for developing a workflow platform based on Argo-Workflow in my company. Therefore, recently I have made some improvements or bug fixes for certain functionalities that we rely on.

I started off the same way (creating a new version of a Batch MLOps platform), so that makes sense! That's how most contributors get started in OSS, if I had to say.

> There are no expectations of you to contribute in the longer term, you're still welcome to become a member and see where that takes you.

Yea as Alan said, still welcome to become a Member!
I actually said [something similar](https://github.com/argoproj/argo-workflows/pull/11352#issuecomment-1634870301) when Terry invited me to become a Member. And now, 6 months later, I'm a top contributor and Approver 😅 

> Would you be interested in working on adding such an Action?

Also, independent of that, would you be interested in working on adding a GH Action to allow `/retest` comments from Members? (although that would mainly benefit you if you become a Member)

**Reply by agilgur5** [reply]
> Also, independent of that, would you be interested in working on adding a GH Action to allow /retest comments from Members? (although that would mainly benefit you if you become a Member)

I wrote up #12864 for this

**Reply by jswxstw** [reply]
I see. I'll create a Membership request later.

> would you be interested in working on adding a GH Action to allow /retest comments from Members

I'm not familiar with GH Actions now, but if it's not urgent, I can learn and try to support it in my spare time.
